tests/extmod/machine_uart_irq_txidle.py

```diff
@@ -0,0 +1,82 @@
+# Test machine.UART.IRQ_TXIDLE firing after a transmission.
+# Does not require any external connections.
+
+try:
+    from machine import UART
+
+    UART.IRQ_TXIDLE
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+# Configure pins based on the target.
+hard_irq = False
+if "rp2" in sys.platform:
+    uart_id = 0
+    tx_pin = "GPIO0"
+    rx_pin = "GPIO1"
+    min_window = 1
+    hard_irq = True
+elif "samd" in sys.platform and "ItsyBitsy M0" in sys.implementation._machine:
+    uart_id = 0
+    tx_pin = "D1"
+    rx_pin = "D0"
+    # For SAMD delay_ms has to be used for the trailing window, and the
+    # mininmal time is 11 ms to allow for scheduling.
+    min_window = 11
+    hard_irq = True
+elif "samd" in sys.platform and "ItsyBitsy M4" in sys.implementation._machine:
+    uart_id = 3
+    tx_pin = "D1"
+    rx_pin = "D0"
+    min_window = 11
+elif "mimxrt" in sys.platform and "Teensy 4" in sys.implementation._machine:
+    uart_id = 1
+    tx_pin = None
+    min_window = 0
+elif "nrf" in sys.platform:
+    uart_id = 0
+    tx_pin = None
+    min_window = 0
+else:
+    print("Please add support for this test on this platform.")
+    raise SystemExit
+
+
+def irq(u):
+    print("IRQ_TXIDLE:", u.irq().flags() == u.IRQ_TXIDLE)
+
+
+text = "Hello World" * 20
+
+# Test that the IRQ is called after the write has completed.
+for bits_per_s in (2400, 9600, 115200):
+    if tx_pin is None:
+        uart = UART(uart_id, bits_per_s)
+    else:
+        uart = UART(uart_id, bits_per_s, tx=tx_pin, rx=rx_pin)
+
+    if hard_irq is True:
+        uart.irq(irq, uart.IRQ_TXIDLE, hard=True)
+    else:
+        uart.irq(irq, uart.IRQ_TXIDLE)
```

Comment:
It should be possible to do `uart.irq(irq, uart.IRQ_TXIDLE, hard=hard_irq)`, unconditionally.

Reply:
The CC200 port for instance does not support the `hard=xxx` argument. But CC3200 does not provide the IRQ_TXIDLE event, so the test will be skipped.

Reply:
Nevertheless changed.